Uchu.World/Scripting/Native/ObjectScript.cs

```diff
@@ -0,0 +1,485 @@
+                    LDFInText = new LegoDataDictionary()
+                    {
+                        {name, value, type}
+                    }.ToString(),
+                });
+            });
+        }
+
+        /// <summary>
+        /// Gets a stored variable of the script.
+        /// </summary>
+        /// <param name="name">Name of the variable to fetch.</param>
+        /// <typeparam name="T">Type of the value.</typeparam>
+        /// <returns>Value of the variable to fetch.</returns>
+        public T GetVar<T>(string name)
+        {
+            return (T) this._variables[name];
+        }
+        #endregion
+
+        #region Timers
+        /// <summary>
+        /// Starts a timer to be called back later. The timer
+        /// can be cancelled.
+        /// </summary>
+        /// <param name="duration"></param>
+        /// <param name="timerName"></param>
+        public void AddTimerWithCancel(float duration, string timerName)
+        {
+            // Cancel the existing timer.
+            this.CancelTimer(timerName);
+            
+            // Create and add the timer.
+            var timer = new Timer(duration * 1000);
+            timer.Elapsed += (sender, args) =>
+            {
+                this.OnTimerDone(timerName);
+            };
+            timer.AutoReset = false;
+            timer.Start();
+            this._timers[timerName] = timer;
+        }
+
+        /// <summary>
+        /// Cancels a timer.
+        /// </summary>
+        /// <param name="timerName">Timer to cancel.</param>
+        public void CancelTimer(string timerName)
+        {
+            if (!this._timers.ContainsKey(timerName)) return;
+            this._timers[timerName].Stop();
+            this._timers.Remove(timerName);
+        }
+
+        /// <summary>
+        /// Cancels all the timers of the script.
+        /// </summary>
+        public void CancelAllTimers()
+        {
+            foreach (var timer in this._timers.Values)
+            {
+                timer.Stop();
+            }
+            this._timers.Clear();
+        }
+
+        /// <summary>
+        /// Callback for the timer completing.
+        /// </summary>
+        /// <param name="timerName">Timer that was completed.</param>
+        public virtual void OnTimerDone(string timerName)
+        {
+            
+        }
+        #endregion
+
+        #region Physics
+        /// <summary>
+        /// Adds a cylinder physics body to the object.
+        /// </summary>
+        /// <param name="radius"></param>
+        /// <param name="name"></param>
+        public void SetProximityRadius(float radius, string name)
+        {
+            // Add the cylinder physics body.
+            // Even if the physics is overriden, the physics body being tied
+            // to a physics component allows it to play nicely in the existing system.
+            var physics = this.GameObject.AddComponent<PhysicsComponent>();
```

Comment:
Instead of overriding the physics and leaving a 'detached' physics body, you can make a new gameobject and attach the physics to that like so:
```cs
var proximityObject = World.GameObject.Instantiate(GameObject.Zone);
var physics = proximityObject.AddComponent<PhysicsComponent>();
```

Reply:
Thanks for providing the example of what to do. The requested change has been made and the script that uses this (the bus door script) still works as intended.